packages/playwright-core/src/server/injected/ariaSnapshot.ts

```diff
@@ -355,16 +363,19 @@ export function renderAriaTree(ariaSnapshot: AriaSnapshot, options?: { mode?: 'r
     }
 
     const escapedKey = indent + '- ' + yamlEscapeKeyIfNeeded(key);
-    if (!ariaNode.children.length) {
+    const hasProps = !!Object.keys(ariaNode.props).length;
+    if (!ariaNode.children.length && !hasProps) {
       lines.push(escapedKey);
-    } else if (ariaNode.children.length === 1 && typeof ariaNode.children[0] === 'string') {
+    } else if (ariaNode.children.length === 1 && typeof ariaNode.children[0] === 'string' && !hasProps) {
       const text = includeText(ariaNode, ariaNode.children[0]) ? renderString(ariaNode.children[0] as string) : null;
       if (text)
         lines.push(escapedKey + ': ' + yamlEscapeValueIfNeeded(text));
       else
         lines.push(escapedKey);
     } else {
       lines.push(escapedKey + ':');
+      for (const [name, value] of Object.entries(ariaNode.props))
+        lines.push(indent + '  - /' + name + ': ' + yamlEscapeValueIfNeeded(value));
```

Comment:
maybe place properties under special child `properties:` and not use the special symbol prefix `/`?